orderer/common/server/main.go

```diff
@@ -124,8 +124,13 @@ func Main() {
 	var clusterDialer *cluster.PredicateDialer
 	var clusterType, reuseGrpcListener bool
 	var serversToUpdate []*comm.GRPCServer
-	if conf.General.BootstrapMethod == "file" {
+	bootstrapMethod := conf.General.BootstrapMethod
+	if bootstrapMethod == "file" || bootstrapMethod == "none" {
 		bootstrapBlock := extractBootstrapBlock(conf)
+		// loop through all channels to obtain system channel when bootstrapmethod is set to none.
+		if bootstrapMethod == "none" {
```

Comment:
Why restrict to the "none" method? Why not also use the extraction from the ledger in the case the extractBootstrapBlock(conf) is nil?

Reply:
good point, done.